source/basic.tex

```diff
@@ -550,6 +550,15 @@
 requirement applies recursively)\footnote{\ref{dcl.fct.default}
 describes how default argument names are looked up.}; and
 
+\item if \tcode{D} is a function that
```

Comment:
This "is a function" restriction is new, and appears to be incorrect for the "invokes a function with a precondition" part. Perhaps:

> if `D` invokes a function with a precondition, or is a function that contains an assertion or has a postcondition, [...]

Reply:
The "pure" model says preconditions are checked in the caller (so we need to account for that), but we have otherwise drafted wording that assumes (under as-if) that preconditions are actually evaluated in the callee. So, I think we should actually care for both places where precondition checking code could be emitted.

Reply:
I agree. However, this appears to be a normative change relative to the wording we were given, so can you send an email to the core reflector to make sure this won't surprise anyone?